enzyme/Enzyme/EnzymeLogic.cpp

```diff
@@ -3078,19 +3255,19 @@ Function *EnzymeLogic::CreatePrimalAndGradient(
   }
 
   if (hasMetadata(key.todiff, "enzyme_gradient")) {
-
-    DIFFE_TYPE subretType = key.todiff->getReturnType()->isFPOrFPVectorTy()
-                                ? DIFFE_TYPE::OUT_DIFF
-                                : DIFFE_TYPE::DUP_ARG;
+    std::set<llvm::Type *> seen;
```

Comment:
```suggestion
    SmallVector<llvm::Type *,4> seen;
```

Reply:
This is required by the api of whatType, not opposed to changing to smallset, but is the calling conv atm.